src/EventStore.Projections.Core/Services/Management/ManagedProjection.cs

```diff
@@ -489,16 +500,13 @@ private readonly
 				DeleteStream(projectionNamesBuilder.GetOrderStreamName(), DeleteIfConditionsAreMet);
 			}
 
-			if (PersistedProjectionState.DeleteCheckpointStream) {
+			if(PersistedProjectionState.CheckpointStreamNeedsDeleted()){
```

Comment:
The `NumberOfPrequisitesMetForDeletion` needs to be updated to not include the CheckpointStreams [here](https://github.com/EventStore/EventStore/blob/master/src/EventStore.Projections.Core/Services/Management/ManagedProjection.cs#L425). Otherwise, OneTime projections get stuck in the deleting state because they're waiting for the checkpoint stream to be deleted.

Reply:
No problem. I will call into the new extension method CheckpointStreamNeedsDeleted in there.

Reply:
Thanks! It looks like there's one last piece missing though in terms of one-time projections - if none of the options are enabled then the `NumberOfPrerequisitesMetForDeletion` starts out as 0, and nothing actually triggers the `DeleteIfConditionsAreMet` function any more, leaving the projection stuck in the deleting state.

I think adding a check for that case in the `DeleteProjectionStreams` method should fix that